File: src/game-logic/deck.js
// src/game-logic/deck.js

const SUITS = ['hearts', 'diams', 'clubs', 'spades'];
const RANKS = ['A', '2', '3', '4', '5', '6', '7', '8', '9', '10', 'J', 'Q', 'K'];

/**
 * Creates a new Loba deck, which consists of two standard 52-card decks
 * plus four jokers.
 * @returns {Array} An array of card objects.
 */
export function createDeck() {
    let deck = [];
    // Two standard decks
    for (let i = 0; i < 2; i++) {
        for (const suit of SUITS) {
            for (const rank of RANKS) {
                deck.push({ suit, rank });
            }
        }
    }
    // Four jokers
    for (let i = 0; i < 4; i++) {
        deck.push({ suit: 'joker', rank: 'Joker' });
    }
    return deck;
}

/**
 * Shuffles a deck of cards in place.
 * @param {Array} deck The deck to shuffle.
 */
export function shuffleDeck(deck) {
    for (let i = deck.length - 1; i > 0; i--) {
        const j = Math.floor(Math.random() * (i + 1));
        [deck[i], deck[j]] = [deck[j], deck[i]];
    }
}

File: python/loba_rl/deck.py
import random
from typing import List
from .card import Card, SUITS, RANKS, JOKER

def create_deck() -> List[Card]:
    """
    Creates a Loba deck, which consists of two standard 52-card decks
    plus four jokers (108 cards total).
    """
    deck = []
    # Two standard decks
    for _ in range(2):
        for suit in SUITS:
            for rank in RANKS:
                deck.append(Card(rank, suit))
    # Four jokers
    for _ in range(4):
        deck.append(JOKER)
    return deck

def shuffle_deck(deck: List[Card]) -> None:
    """
    Shuffles the deck in place.
    """
    random.shuffle(deck)
